src/main/java/org/gridsuite/modification/server/modifications/NetworkModificationApplicator.java

```diff
@@ -122,4 +128,33 @@ private NetworkModificationException handleException(NetworkModificationExceptio
             .build());
         return networkModificationException;
     }
+
+    public static TypedValue computeHighestSeverityFromReporterModel(ReporterModel reporter) {
+        Stream<TypedValue> reportsSeverities = reporter.getReports().stream()
+                .map(NetworkModificationApplicator::computeHighestSeverityFromReport);
+
+        Stream<TypedValue> subReportersSeverities = reporter.getSubReporters().stream()
+                .map(NetworkModificationApplicator::computeHighestSeverityFromReporterModel);
+
+        return Stream.concat(reportsSeverities, subReportersSeverities)
+                .max(new TypedValueComparator()).orElse(TypedValue.TRACE_SEVERITY); // orElse not necessary but we never know
```

Comment:
Could factorize out a `private static TypedValueComparator CMP = new TypedValueComparator();` no ?

Reply:
Good idea I do it